examples/add_points_with_multi_color_text.py

```diff
@@ -0,0 +1,50 @@
+"""
+Display a points layer on top of an image layer using the add_points and
+add_image APIs
+"""
+
+import numpy as np
+import napari
+from napari.layers.utils.property_map import PropertyMap
+
+# add the image
+viewer = napari.view_image(np.zeros((400, 400)))
+# add the points
+points = np.array([[100, 100], [200, 300], [333, 111]])
+
+# create properties for each point
+properties = {
+    'confidence': np.array([1, 0.5, 0]),
+    'good_point': np.array([True, False, False]),
+}
+
+good_point_color = {
+    False: 'green',
+    True: 'blue',
+}
+
+text = {
+    'text': 'Confidence is {confidence:.2f}',
+    'size': 20,
+    'color': PropertyMap.from_discrete_map('good_point', good_point_color),
+    'translation': np.array([-30, 0]),
+}
+
+# create a points layer where the face_color is set by the good_point property
+# and the edge_color is set via a color map (grayscale) on the confidence property.
+points_layer = viewer.add_points(
+    points,
+    properties=properties,
+    text=text,
+    size=20,
+    edge_width=7,
+    edge_color='confidence',
+    edge_colormap='gray',
+    face_color='good_point',
+    face_color_cycle=good_point_color,
```

Comment:
One thing I'd probably like to harmonize is that a color map is a function that maps values in a property to colors, and a color cycle is just a particular such function — one that applies a modulo operation to its input space. In that case, the kwarg should be `face_colormap` here also.

Reply:
Agreed.

Note that in this PR, I am not using `PropertyMap` for `Points.face_color` yet, which is why I must using the existing API to specify the dict using `face_color_cycle`.

By the end of this larger work, I think that a good goal is that there is only `face_color` (not `face_color_cycle` nor `face_colormap`) for `TextManager`, `Points`, `Shapes`, etc., and that field always accepts the same types.